packages/components/src/progressive-disclosure-panel/item.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * Internal dependencies
+ */
+import { usePanelContext } from './';
+
+// This wraps controls to be conditionally displayed within a progressive
+// disclosure panel. It helps prevent props being applied to HTML elements that
+// would otherwise be invalid.
+const ProgressiveDisclosurePanelItem = ( { children, label } ) => {
```

Comment:
The `label` prop is dual purpose. 

In the main `ProgressiveDisclosurePanel` the label prop is used to generate the menu items. As such, the `label` is actually used as the human-readable text when they are rendered via the inner `ProgressiveDisclosurePanelTitle`. 

It didn't make sense to me that these panel's would have menu items with the same label, so the label could also serve as the key to access whether the menu items display status. This avoided the consumer needing to add an additional prop to each item, more often than not almost the same as the label. It also greatly simplified the code.

Do you still see this as something that needs changing?

Reply:
I see, thank you for the explanation!

If we keep `label` as-is, maybe it would a good idea to add a few comments in the code to make its dual purpose more clear? E.g.:

- A comment in the README (and in the future JSDoc of the `types.ts` file) saying that `label` is supposed to be unique amongst the different `ProgressiveDisclosurePanelItem`s
- A comment in `packages/components/src/progressive-disclosure-panel/item.js` where `menuItems[ label ]` is accessed, explaining that `label` is used as a key in `ProgressiveDisclosurePanel` ?
- Optionally, an extra check in the `ProgressiveDisclosurePanel` component to make sure that all of the "valid" `ProgressiveDisclosurePanelItem`s all have unique labels

What do you think?

Reply:
Thanks, I've updated the current docs and added an inline comment within the panel item component to reflect the nature and use of the `label` prop. 

I haven't yet written any code to enforce uniqueness of labels. I'm happy to do so as needed once the remaining rework of the component is completed.